### PR TITLE
 Implement `plot_optimization_history` and `_get_optimization_history_plot` with Matplotlib backend.

### PR DESCRIPTION
**Pull request by ytknzw**
This PR proposes to implement a ``matplotlib`` backend for plotting optimization history based on PR #1756 .

## Motivation
To provide another visualization option for users, who have only ``plotly`` now.
See issue #1539 and PR #1707 for the original motivation.

## Description of the changes
Create  ``optuna.visualization.matplotlib._optimization_history.plot_optimization_history`` and ``optuna.visualization.matplotlib._optimization_history._get_optimization_history_plot``.

### CODE REVIEW

**Review by crcrpar — COMMENTED** [review]
Generally it looks good including the output.

image

(This figure is generated using an edited visualization example jupyter notebook.)

File: optuna/visualization/matplotlib/_optimization_history.py [hunk 1]
```diff
@@ -1,11 +1,79 @@
...
+
+    _imports.check()
+    return _get_optimization_history_plot(study)
+
+
+def _get_optimization_history_plot(study: Study) -> "Axes":
+
+    # Set up the graph style.
+    plt.style.use("ggplot")  # Use ggplot style sheet for similar outputs to plotly.
+    fig, ax = plt.subplots()
```
Comment (toshihikoyanase) [inline optuna/visualization/matplotlib/_optimization_history.py:40]: A nit, but I guess `fig` is not used in this function.
```suggestion
    _, ax = plt.subplots()
```

**Review by toshihikoyanase — APPROVED** [review]
Thank you for your PR. LGTM with a nit.

File: optuna/visualization/matplotlib/_optimization_history.py (hunk 1 above)
Comment (ytknzw) [reply optuna/visualization/matplotlib/_optimization_history.py]: Thank you for reviewing! Removed it.

Comment (toshihikoyanase) [reply optuna/visualization/matplotlib/_optimization_history.py]: Thank you for your swift action. LGTM again.

**Review by crcrpar — APPROVED** [review]

**Pull request merged**